### PR TITLE
completed js data model | Mohamed Elmessiry

### PR DESCRIPTION
**Pull request by Mohamed-Elmessiry**
Screen Shot 2021-01-26 at 4 28 21 PM
Screen Shot 2021-01-26 at 4 28 29 PM

### CODE REVIEW

File: javascript-data-model/order-history.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+var orderHistory = [
+  order1 = [
```
Comment (codingcodymiller) [inline javascript-data-model/order-history.js:2]: This should be removed, you don't individually name data in arrays.

File: javascript-data-model/order-history.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      shipTo: 'JS Masher',
+      order: '114-3941689-877232',
+      deliveredDate: 'Aug 8, 2020',
+      bookTitle: 'Javscript for Impantient programmers',
+      bookAuthor: 'Rauschmayer, Dr.Axel',
+      returnWindowClosingDate: 'Sep 7, 2020',
+      price: '$31.55'
+    }
+  ],
+  order2 = [
```
Comment (codingcodymiller) [inline javascript-data-model/order-history.js:15]: This should be removed, you don't individually name data in arrays.

File: javascript-data-model/order-history.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      shipTo: 'JS Masher',
+      order: '113-9984268-1280257',
+      deliveredDate: 'Jul 20,2020',
+      bookTitle: 'The Timeless Way of Building',
+      bookAuthor: 'Alexander, Christopher',
+      returnWindowClosingDate: 'Aug 19 2020',
+      price: '$41.33'
+    }
+  ],
+  order3 = [
```
Comment (codingcodymiller) [inline javascript-data-model/order-history.js:28]: This should be removed, you don't individually name data in arrays.

File: javascript-data-model/order-history.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      total: '$17.22',
+      shipTo: 'JS Masher',
+      order: '114-2875557-9059049',
+      deliveredDate: 'Jul 7,2020',
+      itemName: 'Gamecube Controller Adapter, Super Smash Bros Switch Gamcube Adapter for WII U,PC Support Turbo and Vibration Features. No Driver and No Lag-Gamecube Adapter',
+      returnWindowClosingDate: 'Aug 5, 2020',
+      price: '$15.98'
+    }
+  ],
+  order4 = [
```
Comment (codingcodymiller) [inline javascript-data-model/order-history.js:40]: This should be removed, you don't individually name data in arrays.

File: javascript-data-model/order-history.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    {
+      order: '113-2883177-2648248',
+      deliveredDate: 'July 3, 2020',
+      bookTitle: 'The Art of Sql',
+      bookAuthor: 'Faroult Stephane',
+      returnWindowClosingDate: 'Aug 4,2020',
+      price: '$33.99'
+
+    }
+  ]
```
Comment (codingcodymiller) [inline javascript-data-model/order-history.js:62]: Combine all this into one object with a similar structure to this:

```
{
    orderDate: 20200703,
    shipTo: 'JS Masher',
    orderNumber: 11328831772648248,
    items:
    [
      {
        name: 'Gamecube Controller - Super Smash Bros. Edition (Nintendo Switch)',
        price: 94.95
      },
      {
        deliveryDate: 20200705,
        name: 'The Art of Sql',
        price: 33.99
      }
    ]
  }
```

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Good work so far! Please see my notes

File: javascript-data-model/order-history.js (hunk 5 above)
Comment (Mohamed-Elmessiry) [reply javascript-data-model/order-history.js]: Screen Shot 2021-01-28 at 3 18 35 PM
Screen Shot 2021-01-28 at 3 18 44 PM

File: javascript-data-model/order-history.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 var orderHistory = [
-  order1 = [
+  [
```
Comment (codingcodymiller) [inline javascript-data-model/order-history.js:2]: This is still incorrect, you don't need to wrap each individual order object in an array, there is already an array to hold all orders in the `orderHistory` variable.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Another note for you:

File: javascript-data-model/order-history.js (hunk 6 above)
Comment (Mohamed-Elmessiry) [reply javascript-data-model/order-history.js]: Screen Shot 2021-01-28 at 4 30 50 PM
Screen Shot 2021-01-28 at 4 30 56 PM

File: javascript-data-model/order-history.js [hunk 7]
```diff
@@ -1,66 +1,60 @@
...
+        bookTitle: 'The Art of Sql',
+        bookAuthor: 'Faroult Stephane',
+        returnWindowClosingDate: 'Aug 4,2020',
+        price: '$33.99'
 
-  ]
+      }
+
+    ]
+  }
```
Comment (codingcodymiller) [inline javascript-data-model/order-history.js]: Your other objects should match this format.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: javascript-data-model/order-history.js (hunk 7 above)
Comment (Mohamed-Elmessiry) [reply javascript-data-model/order-history.js]: Screen Shot 2021-01-28 at 6 01 10 PM
Screen Shot 2021-01-28 at 6 01 17 PM

**Review by codingcodymiller — APPROVED** [review]
Awesome work! 🎉

**Pull request merged**